File: src/main/java/cn/edu/tsinghua/sdfs/protocol/FilePacket.java
package cn.edu.tsinghua.sdfs.protocol;


import java.io.File;

import static cn.edu.tsinghua.sdfs.protocol.command.Command.FILE_PACKET;

public class FilePacket extends Packet {

    private File file;

    @Override
    public int getCommand() {
        return FILE_PACKET;
    }

    public FilePacket(File file) {
        this.file = file;
    }

    public File getFile() {
        return file;
    }

}

File: src/main/java/cn/edu/tsinghua/sdfs/protocol/FilePacket.kt
package cn.edu.tsinghua.sdfs.protocol


import cn.edu.tsinghua.sdfs.protocol.command.Command.Companion.FILE_PACKET
import java.io.File

data class FilePacket(var file: File?) : Packet() {


    constructor() : this(null)

    override val command: Int
        get() = FILE_PACKET

}
